librecad/src/actions/rs_actiondefault.cpp

```diff
@@ -109,16 +111,152 @@ void RS_ActionDefault::keyReleaseEvent(QKeyEvent* e) {
 }
 
 
+/*
+   Highlights hovered entities that are visible and not locked.
+
+   - by Melwyn Francis Carlo <carlo.melwyn@outlook.com>
+*/
+void RS_ActionDefault::highlightHoveredEntities(const RS_Vector& currentMousePosition)
+{
+    for (auto entity : *graphicView->getContainer())
+    {
+        if (entity != nullptr)
+        {
+            if (entity->isVisible() && ! entity->isLocked() && (entity->getHighlightedEntityParent() == nullptr))
+            {
+                const double hoverToleranceFactor = (entity->rtti() == RS2::EntityEllipse) 
+                                                  ? hoverToleranceFactor1 
+                                                  : hoverToleranceFactor2;
+
+                const double hoverTolerance { hoverToleranceFactor / graphicView->getFactor().magnitude() };
+
+                const double hoverTolerance_adjusted = ((entity->rtti() != RS2::EntityEllipse) && (hoverTolerance < minimumHoverTolerance)) 
+                                                     ? minimumHoverTolerance 
+                                                     : hoverTolerance;
+
+                bool isPointOnEntity = false;
+
+                if (((entity->rtti() >= RS2::EntityDimAligned) && (entity->rtti() <= RS2::EntityDimLeader)) 
+                ||   (entity->rtti() == RS2::EntityText)       || (entity->rtti() == RS2::EntityMText))
+                {
+                    double nearestDistanceTo_pointOnEntity;
+
+                    RS_Vector dummyVector = entity->getNearestPointOnEntity(currentMousePosition, true, &nearestDistanceTo_pointOnEntity);
+
+                    if (dummyVector) { /* This is a dummy code to suppress the 'unused variable' compiler warning. */ }
+
+                    if (nearestDistanceTo_pointOnEntity <= hoverTolerance_adjusted) isPointOnEntity = true;
+                }
+                else
+                {
+                    isPointOnEntity = entity->isPointOnEntity(currentMousePosition, hoverTolerance_adjusted);
+                }
+
+                if (isPointOnEntity)
+                {
+                    RS_Color whiteColor(QColor(Qt::white));
```

Comment:
In here the compiler gives an error says qualified name is not allowed, when i changed it to QColor(255,255,255) it worked.